airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeV1V2Migrator.java

```diff
@@ -0,0 +1,93 @@
+package io.airbyte.integrations.destination.snowflake.typing_deduping;
+
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.integrations.base.destination.typing_deduping.BaseDestinationV1V2Migrator;
+import io.airbyte.integrations.base.destination.typing_deduping.CollectionUtils;
+import io.airbyte.integrations.base.destination.typing_deduping.NamespacedTableName;
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig;
+import io.airbyte.integrations.destination.NamingConventionTransformer;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Optional;
+import lombok.SneakyThrows;
+
+public class SnowflakeV1V2Migrator extends BaseDestinationV1V2Migrator<SnowflakeTableDefinition> {
+
+  private final NamingConventionTransformer namingConventionTransformer;
+
+  private final JdbcDatabase database;
+
+  private final String databaseName;
+
+  public SnowflakeV1V2Migrator(final NamingConventionTransformer namingConventionTransformer, final JdbcDatabase database,
+                               final String databaseName) {
+    this.namingConventionTransformer = namingConventionTransformer;
+    this.database = database;
+    this.databaseName = databaseName;
+  }
+
+  @SneakyThrows
+  @Override
+  protected boolean doesAirbyteInternalNamespaceExist(final StreamConfig streamConfig) {
+    return database
+        .queryJsons(
+            """
+                SELECT SCHEMA_NAME
+                FROM information_schema.schemata
+                WHERE schema_name = ?;
```

Comment:
I think we also need to check `AND table_catalog = ?`, `databaseName`

Reply:
good call, slightly different syntax of `AND catalog_name = ?` , `databaseName`. I also missed a `!` at the beginning of the expression 🤦